Assets/Scripts/Player/PlayerProgress.cs

```diff
@@ -110,14 +112,15 @@ public void Respawn()
             GameManager.RestartLevel();
             playerGhostRun.RestartRun();
             ResetCollectibles();
-            ResetCheckpoints();

```

Comment:
might as well move everything in the if statement over

Reply:
Do you want me to create a new function to handle this reset and apply it only on level reset? Or are you asking for something else? For example right now we remove portals if you respawn without having reached the second checkpoint, moving this code to its own function would change that. It may be out of the scope of this task, but if you know what we want to do I can implement it.